### PR TITLE
improve(pr-summary): rewrite skill for higher PR-process fidelity

### PR DESCRIPTION
**Pull request by moxious**
## Summary

Rewrite the `/pr-summary` skill so it captures the "why" content the diff cannot supply, makes its links permanent, and pairs cleanly with `/review`. Retires `/tidy-up`, which the pre-commit hook already covered.

## What to look at

- [`.cursor/skills/pr-summary/SKILL.md`](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/.cursor/skills/pr-summary/SKILL.md) — full rewrite. Two load-bearing additions: the [nine hard constraints block](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/.cursor/skills/pr-summary/SKILL.md#L10-L22), which the skill enforces over any other instruction, and the [Phase 0.5 author interview](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/.cursor/skills/pr-summary/SKILL.md#L102-L120) — three calibrated questions that fire only when commit bodies are thin and `--quick` was not invoked. Agents read the skill on every invocation, so its size is a per-run context tax; the rewrite stays at ~265 lines on purpose.
- [`.cursor/skills/tidy-up/SKILL.md`](https://github.com/grafana/grafana-pathfinder-app/blob/7879b439/.cursor/skills/tidy-up/SKILL.md) — deleted. The pre-commit hook (lint-staged + Prettier + typecheck) and `npm run check` already enforce everything it encoded; the audit found no judgment calls that required human-curated guidance.
- [`CLAUDE.md`](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/CLAUDE.md) — drops the `tidy-up` row from the skills table and updates the `pr-summary` row to mention the interview and `--quick`.
- [`docs/_maintenance-backlog.md`](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/docs/_maintenance-backlog.md) — removes the now-stale exclusion entry for the deleted tidy-up skill.

## Why

The previous `/pr-summary` had three concrete failure modes:

1. It mirrored the diff (the "What changed" section was a prose `git diff --stat`) instead of guiding reviewer attention to the load-bearing interactions.
2. It polluted "Test plan" with CI-enforced commands (`npm run check`, `npm run typecheck`), which add no signal — CI fails the PR if any of those fail, so listing them was pure noise.
3. It had no mechanism for capturing the "why" content only the author holds. If commit bodies were thin, the skill emitted a placeholder and shipped.

The rewrite addresses each: "What to look at" replaces "What changed" with attention-routing guidance; [constraint #7](https://github.com/grafana/grafana-pathfinder-app/blob/0e1c6e7b064227a60206cc846ba25389225c6edd/.cursor/skills/pr-summary/SKILL.md#L20) bans CI commands explicitly; Phase 0.5 asks the three questions calibrated to surface diff-invisible context (problem, rejected alternatives, reviewer gotchas).

The `/tidy-up` retirement is part of the same fidelity push — the skills directory should encode judgment, not wrap hooks that already run automatically.

## How to verify

1. On a branch with thin commit bodies, run `/pr-summary`. Confirm Phase 0.5 fires with the three verbatim questions before drafting anything.
2. On a branch with substantive commit bodies (e.g., this one), run `/pr-summary`. Confirm the interview is skipped and the skill goes straight to drafting.
3. Run `/pr-summary --quick`. Confirm the interview is skipped regardless of commit-body depth.
4. On any branch with no PR yet, run `/pr-summary --apply`. Confirm the exit guard fires with "No PR found for this branch. Use `--open` to create one." (folded into Phase 4 step 1).
5. Inspect any draft the skill produces. Confirm every file link uses `https://github.com/grafana/grafana-pathfinder-app/blob/<40-char-SHA>/...`, never `blob/main/` or `blob/<branch>/`.
6. On a synthetic `renovate/` or `dependabot/` branch, run `/pr-summary` and confirm Phase 0 step 1 exits immediately with a one-line `chore(deps):` draft — no full workflow run.

## Breaking changes

- The `/tidy-up` skill is removed. Authors or external docs that referenced it should run `npm run check` directly, or rely on the pre-commit hook (the de-facto behavior already — the skill was rarely invoked explicitly).
- The `/pr-summary` skill's default now includes a three-question interview when commit bodies are thin. Authors who want the old "just draft" behavior should use `/pr-summary --quick`. Combinable with `--apply` and `--open`.

## Out of scope

- i18n reminder in How to verify — surfaced as a small fidelity gap during review; not taken on this branch.
- Draft-vs-diff drift check — surfaced during review as a way to catch drafts that miss a public surface (new env var, new HTTP route); not taken on this branch.
- Reinforced Q3 follow-through rule in Phase 2 — left at its current two-sentence form scattered across phases.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .cursor/skills/pr-summary/SKILL.md [hunk 1]
```diff
@@ -88,9 +111,27 @@ Different change shapes use different sections — match depth to weight:
 
    The `git log` output preserves commit bodies so you can extract motivation.
 
-3. Read the **most recent commit body** in full — it usually contains the substantive `Why` content.
+5. Read the **most recent commit body** in full — it sometimes contains the substantive `Why` content. A commit body is substantive when it contains ≥2 sentences of motivation text beyond the subject line (excluding `Co-authored-by:` trailers and similar footers).
```
Comment (leslie-heinzen) [inline .cursor/skills/pr-summary/SKILL.md]: Commenting on the existing "most recent commit body" phrasing: Kind of wonder if this adds too much weight to the final commit message of multi-commit branches, where the last commit has just partial information. I think `4.` above probably gives the model the context the understand that though.

File: .cursor/skills/pr-summary/SKILL.md [hunk 2]
```diff
@@ -112,45 +153,73 @@ Different change shapes use different sections — match depth to weight:
...
 
-- Read the most recent commit body in full. Extract the substantive motivation paragraph.
-- If linked to a GitHub issue (commit body contains `Fixes #NNN` or `Refs #MMM`), read the issue title and add it as context.
-- If neither commit body nor issue gives clear motivation, emit `[FILL IN: motivation]` and let the author finish. **Do not fabricate.**
-- For `fix:` PRs, this section is usually unnecessary — the Summary already explains the problem.
+- Primary source: answers from Phase 0.5 questions 1 and 2.
+- Secondary source: commit bodies from Phase 0.
+- If linked to a GitHub issue (`Fixes #NNN` / `Refs #MMM`), read the issue title and add it as context.
+- If none of the above provide clear motivation, emit `[FILL IN: motivation]` and let the author finish. **Do not fabricate.**
+- For `fix:` PRs where Summary already states the problem clearly, omit this section.
```
Comment (leslie-heinzen) [inline .cursor/skills/pr-summary/SKILL.md]: Wonder if this might be sufficient for some `chore` PRs too. 🤔 Though I suppose that is what quick mode is for.

File: .cursor/skills/pr-summary/SKILL.md [hunk 3]
```diff
@@ -1,52 +1,64 @@
...
+5. **No unrelated boilerplate.** A typo fix doesn't need a Reversibility section. A two-line fix doesn't need a subsystem table. Match section depth to change weight.
 6. **No source code edits.** This skill only reads — and at most calls `gh pr edit`.
+7. **No CI-enforced commands in How to verify.** CI auto-runs typecheck, lint, prettier, and the test suite on every PR. Do not list `npm run check`, `npm run typecheck`, `npm run test:ci`, or `npm run lint` — only PR-specific manual verification (repro steps, UI interactions, edge cases, integration scenarios CI cannot exercise).
+8. **All file links use the head-commit SHA, not a branch or `main`.** Format: `$REMOTE_URL/blob/$HEAD_SHA/path/to/file`. Never write `../blob/<branch>/...` or `.../blob/main/...` — these go stale the moment the branch is deleted or the file moves.
+9. **No bare artifact codes.** Never write "per A1", "see QC8", or any other section code in isolation. If you link to a design doc or reference a named section, include a one-sentence inline description of what it says. The PR body must be self-standing without requiring the reader to follow the link to parse the sentence.
```
Comment (leslie-heinzen) [inline .cursor/skills/pr-summary/SKILL.md:22]: Love it.

File: .cursor/skills/pr-summary/SKILL.md [hunk 4]
```diff
@@ -1,52 +1,64 @@
...
 
 ## Operating modes
 
 - **Draft mode** (default): print the draft body in a fenced block. Author copy-pastes into the PR description manually. No tool side effects.
-- **Apply mode**: if the user invokes `/pr-summary --apply` or replies "apply" after seeing the draft, write the body to a temp file and call `gh pr edit <number> --body-file <tmp>`. Confirm the PR number from `gh pr view --json number`.
+- **Apply mode** (`--apply`): if the user invokes `/pr-summary --apply` or replies "apply" after seeing the draft, write the body to a temp file and call `gh pr edit <number> --body-file <tmp>`. Confirm the PR number from `gh pr view --json number`. Requires an existing PR.
+- **Open mode** (`--open`): draft as normal, then on confirmation push the branch (if not already pushed) and call `gh pr create` with the title and body. Use when no PR exists yet and you want the skill to open it.
```
Comment (leslie-heinzen) [inline .cursor/skills/pr-summary/SKILL.md]: We chatted about git history rewrites as a possible inclusion here. However, I'm not really sure about it after consdering, because:

* we're a squash merge project, so the value of individual commit messages is diminished.
* history rewrites with rebase can cause a lot of friction sometimes (i.e., the branch has merge commits from main pulled in already, is being collaborated w/multiple devs, etc). Unsure it would be worth it given the previous point.

That said, I personally like the idea and I rebase all the time anyway (😆), so I'll throw this out there for you to consider. Maybe an optional additional `tidy-history` flag specifically for open mode, a mode where the branch is likely not to have issues:

```suggestion
- **Open mode** (`--open`): draft as normal, then on confirmation push the branch (if not already pushed) and call `gh pr create` with the title and body. Use when no PR exists yet and you want the skill to open it.
  - With `--tidy-history`: before pushing, rewrite commits that lack a conventional-commit prefix. For each commit whose subject does not start with `feat:`/`fix:`/`refactor:`/`chore:`/`docs:`/`test:`/`perf:` (or a scoped variant like `feat(scope):`), infer the prefix from the diff and prepend it to the subject.

    Mechanism: run `git rebase <base> --exec '<amend-script>'` where `<amend-script>` inspects `git log -1 --pretty=%s`, checks for a conventional-commit prefix, and runs `git commit --amend -m "<prefix>: <subject>"` when one is missing. The rebase runs non-interactively; if the exec fails on any commit, the rebase halts and the user can `git rebase --abort` to recover (the backup ref is the deeper safety net).

    Safeguards (non-negotiable):
    - Refuse to run if the working tree is dirty (`git status --porcelain` non-empty) or a rebase/merge is in progress.
    - Create a backup ref before rewriting: `git update-ref refs/backup/tidy-<branch>-<unix-ts> HEAD`. Print the ref so the user can recover with `git reset --hard <backup>`.
    - If the branch is already pushed, push with `git push --force-with-lease` after the rewrite. Never plain `--force`.
```

Comment (leslie-heinzen) [reply .cursor/skills/pr-summary/SKILL.md]: If we do something like this, I think we need to also update the "hard constraints" section earlier in the skill to include `git push` and `git rebase`.

**Review by leslie-heinzen — COMMENTED** [review]
@moxious These changes look good, just a couple of questions/thoughts, and an idea for your consideration that I myself am not 100% sold on.

File: .cursor/skills/pr-summary/SKILL.md (hunk 1 above)
Comment (moxious) [reply .cursor/skills/pr-summary/SKILL.md]: I have expanded this to "read the commit bodies" and look for "why" information, since the agent does have to do this anyway.  I think the extra bullet is needed, because the previous bullets tell it to read the commit log, but not what to look for / why.

**Comment by leslie-heinzen** [comment]
@moxious I tested the updates out on a branch. Specifically the `--open --tidy-history` flags: https://github.com/grafana/grafana-pathfinder-app/pull/956

The commits there originally had no prefixes (but detailed bodies). Interestingly the agent presented the re-written message proposal and asked to confirm before doing the rebase, which we didn't specify but is nice. I think it inferred based on the wording about confirming before mutating anything related to the PR body (Phase 3):

> Before I run tidy-history and push, here's what I plan to do for --tidy-history — the 5 strict-mode commits all lack a conventional-commit prefix:

```
| Short hash | Current subject                                        | Proposed rewrite                                                                 |
| ---------- | ------------------------------------------------------ | -------------------------------------------------------------------------------- |
| `179bd039` | add isHighQualitySelectorMethod predicate              | `feat(dom): add isHighQualitySelectorMethod predicate`                           |
| `80a2c64f` | gate fragile selectors in strict mode                  | `feat(recorder): gate fragile selectors in strict mode`                          |
| `a8cd2e82` | strict mode toggle in record overlay                   | `feat(block-editor): strict mode toggle in record overlay`                       |
| `8b0b0588` | drop rejectedCount surface                             | `refactor(recorder): drop rejectedCount surface`                                 |
| `27f50cfe` | replace strict-mode blocking with on-stop confirmation | `refactor(block-editor): replace strict-mode blocking with on-stop confirmation` |
```

> The pr-summary commits (7879b439..0edeecf9) will be left untouched per your instruction.
>
> Apply this draft and open the PR (tidy-history + push --force-with-lease + gh pr create)? Reply "yes" / "apply" to proceed, or tell me what to adjust.

File: .cursor/skills/pr-summary/SKILL.md [hunk 5]
```diff
@@ -1,52 +1,73 @@
...
```
Comment (leslie-heinzen) [inline .cursor/skills/pr-summary/SKILL.md]: Opinion: I always like to open as draft and have a look at CI first before marking the PR as ready and tagging the code owners.

How do we feel about the `--draft` option for the create command?

```suggestion
- **Open mode** (`--open`): draft as normal, then on confirmation push the branch (if not already pushed) and call `gh pr create --draft` with the title and body. Use when no PR exists yet and you want the skill to open it.
```

**Review by leslie-heinzen — APPROVED** [review]
:shipit:
